### PR TITLE
Add hidden flag `--dart-flags` to run command

### PR DESCRIPTION
**Pull request by johnmccutchan**
This is needed so that VM engineers can run with arbitrary debug flags

### CODE REVIEW

**Comment by johnmccutchan** [comment]
Pairs with: Related: https://github.com/flutter/engine/pull/2503

**Comment by johnmccutchan** [comment]
@mhausner

**Comment by johnmccutchan** [comment]
@devoncarew

File: packages/flutter_tools/lib/src/ios/devices.dart [hunk 1]
```diff
@@ -157,7 +157,8 @@ class IOSDevice extends Device {
     bool clearLogs: false,
     bool startPaused: false,
     int debugPort: observatoryDefaultPort,
-    Map<String, dynamic> platformArgs
+    Map<String, dynamic> platformArgs,
+    String dartFlags,
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/ios/devices.dart]: perhaps add a TODO: below

File: packages/flutter_tools/lib/src/ios/simulators.dart [hunk 2]
```diff
@@ -445,6 +446,10 @@ class IOSSimulator extends Device {
     if (debugPort != observatoryDefaultPort)
       args.add("--observatory-port=$debugPort");
 
+    if (dartFlags != null) {
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/ios/simulators.dart]: no curlies needed

**Comment by devoncarew** [comment]
lgtm modulo travis lint issues

File: packages/flutter_tools/lib/src/ios/devices.dart (hunk 1 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/ios/devices.dart]: Done.

File: packages/flutter_tools/lib/src/ios/simulators.dart (hunk 2 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/ios/simulators.dart]: Done.

File: packages/flutter_tools/lib/src/commands/run.dart [hunk 3]
```diff
@@ -41,6 +41,8 @@ abstract class RunCommandBase extends FlutterCommand {
         negatable: true,
         defaultsTo: false,
         help: 'Start tracing during startup.');
+    // Hidden flag for passing options to the VM.
+    argParser.addOption('dart-flags', hide:true);
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/run.dart]: iirc, --verbose will show this option. So we should still have help text.

Also, space after colon.

File: packages/flutter_tools/lib/src/ios/devices.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter_tools/lib/src/ios/devices.dart]: I think the travis errors are just the extraneous comma on the `String dartFlags,` line here.

**Comment by Hixie** [comment]
@johnmccutchan What's the status of this PR?

**Comment by Hixie** [comment]
Looks like it's blocked on https://github.com/flutter/engine/pull/2503

**Comment by Hixie** [comment]
Closing for now, @johnmccutchan will reopen it when he's ready.

**Pull request closed without merging**